mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,589 @@
+        else:
+            self.weightList = None
+
+        self.filePrefix = f'histogram_{mainRunName}'
+
+        baseDirectory = build_config_full_path(
+            config, 'output', 'histogramSubdirectory')
+        if not os.path.exists(baseDirectory):
+            make_directories(baseDirectory)
+
+        self.obsList = config.getexpression(self.taskName, 'obsList')
+        obsDicts = {
+            'AVISO': {
+                'suffix': 'AVISO',
+                'gridName': 'Global_1.0x1.0degree',
+                'gridFileName': 'SSH/zos_AVISO_L4_199210-201012_20180710.nc',
+                'lonVar': 'lon',
+                'latVar': 'lat',
+                'sshVar': 'zos',
+                'pressureAdjustedSSHVar': 'zos'}}
+
+        for regionGroup in self.regionGroups:
+            groupObsDicts = {}
+            mpasMasksSubtask = regionMasksTask.add_mask_subtask(
+                regionGroup=regionGroup)
+            regionNames = mpasMasksSubtask.expand_region_names(
+                self.regionNames)
+
+            # Add mask subtasks for observations and prep groupObsDicts
+            # groupObsDicts is a subsetted version of localObsDicts with an
+            # additional attribute for the maskTask
+            for obsName in self.obsList:
+                localObsDict = dict(obsDicts[obsName])
+                obsFileName = build_obs_path(
+                    config, component=self.componentName,
+                    relativePath=localObsDict['gridFileName'])
+                obsMasksSubtask = regionMasksTask.add_mask_subtask(
+                    regionGroup, obsFileName=obsFileName,
+                    lonVar=localObsDict['lonVar'],
+                    latVar=localObsDict['latVar'],
+                    meshName=localObsDict['gridName'])
+                localObsDict['maskTask'] = obsMasksSubtask
+                groupObsDicts[obsName] = localObsDict
+
+            for regionName in regionNames:
+                sectionName = None
+
+                # Compute weights for histogram
+                if self.weightList is not None:
+                    computeWeightsSubtask = ComputeHistogramWeightsSubtask(
+                        self, regionName, mpasMasksSubtask, self.filePrefix,
+                        self.variableList, self.weightList)
+                    self.add_subtask(computeWeightsSubtask)
+
+                for season in self.seasons:
+
+                    # Generate histogram plots
+                    plotRegionSubtask = PlotRegionHistogramSubtask(
+                        self, regionGroup, regionName, controlConfig,
+                        sectionName, self.filePrefix, mpasClimatologyTask,
+                        mpasMasksSubtask, obsMasksSubtask, groupObsDicts,
+                        self.variableList, self.weightList, season)
+                    plotRegionSubtask.run_after(mpasMasksSubtask)
+                    plotRegionSubtask.run_after(obsMasksSubtask)
```

Comment:
```suggestion
                    plotRegionSubtask.run_after(mpasMasksSubtask)
                    plotRegionSubtask.run_after(obsMasksSubtask)
                    plotRegionSubtask.run_after(computeWeightsSubtask)
```
Since the plotting depends on the weights, you need to make sure you don't try to plot before you have the weights you need.